src/Shared/ReuseableStringBuilder.cs

```diff
@@ -186,6 +186,35 @@ internal ReuseableStringBuilder Append(string value, int startIndex, int count)
             return this;
         }
 
+        public ReuseableStringBuilder AppendJoined(char separator, ICollection<string> strings)
+        {
+            LazyPrepare();
+            _cachedString = null;
+
+            var i = 0;
+
+            foreach (var s in strings)
+            {
+                _borrowedBuilder.Append(s);
+
+                i++;
+                if (i < strings.Count - 1)
```

Comment:
extreme nitpick: Maybe you could count _down_ from `int separatorsRemaining = strings.Count - 1` to avoid having to get the count multiple times and simplify the condition?